spec/gp_update_metadata_source_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::GpUpdateMetadataSourceAction do
+  it 'updates any block in a given .po file with the values from the given sources' do
+    Dir.mktmpdir do |dir|
+      # 1: Create a dummy .po file to use as input.
+      output_path = File.join(dir, 'output.po')
+      dummy_text = <<~PO
+        msgctxt "v1.0-whats-new"
+        msgid "this value should change"
+        msgstr ""
+        msgctxt "release_note_0122"
+        msgid "previous version notes required to have current one added"
+        msgstr ""
+        msgctxt "key1"
+        msgid "this value should change"
+        msgstr ""
+        msgctxt "key2"
+        msgid "this value should change"
+        msgstr ""
+      PO
+      File.write(output_path, dummy_text)
+
+      # 2: Create source files with value to insert in the .po
+      release_notes_path = File.join(dir, 'release_notes.txt')
+      File.write(release_notes_path, "- release notes\n- more release notes")
+      whats_new_path = File.join(dir, 'whats_new.txt')
+      File.write(whats_new_path, "- something new\n- something else new")
+      file_1_path = File.join(dir, '1.txt')
+      File.write(file_1_path, 'value 1')
+      file_2_path = File.join(dir, '2.txt')
+      File.write(file_2_path, 'value 2')
+      file_3_path = File.join(dir, '3.txt')
+      File.write(file_3_path, 'value 3')
+
+      described_class.run(
+        po_file_path: output_path,
+        release_version: '1.23',
+        source_files: {
+          release_note: release_notes_path,
+          whats_new: whats_new_path,
```

Comment:
Wait what's the difference between `release_note` and `whats_new`? Or is it an intentional duplication you made here to be sure to cover both cases?

I mean I see that we indeed have both types of `ReleaseNoteMetadataBlock` and `WhatsNewMetadataBlock` in the current/existing codebase, to handle them separately, but I can't understand if there's supposed to be a difference in meaning between the two keys, or if, like I fear it is, it's just 2 names for the same thing — aka the Release Notes entry — and it's just that we've historically used different key names for those depending on the projects, in which case even if it's good to have unit tests that check the case for when both are present, in practice it will only be one or the other, depending on the key that the client project historically uses?

In that case in the future refactoring you plan to work on, let's only keep one and formalize its usage on all clients 🙂 

Related: see also [my recent comment on this WCiOS PR here](https://github.com/woocommerce/woocommerce-ios/pull/5906#discussion_r785287011) about this.

Reply:
> Wait what's the difference between `release_note` and `whats_new`?

I think the code expects `release_note` for Android release notes and `whats_new` for the iOS ones. They end up matching different `msgctxt` values and have different handling, i.e. for Android it keeps the `n - 1` version while for iOS it always keeps only the latest.

> In that case in the future refactoring you plan to work on, let's only keep one and formalize its usage on all clients 🙂

I'd like to get to that place, yeah, even though it might require changing how the process works. I don't see the value in keeping the `n - 1` info in the Android `.po`, for example. If we dropped that, and allowed for either Android or iOS adopting the other's `msgctxt`, we should be able to consolidate.